brian2/input/spikegeneratorgroup.py

```diff
@@ -150,15 +150,23 @@ def before_run(self, run_namespace):
             if period < dt:
                 raise ValueError('The period of %s is %s, which is smaller '
                                  'than its dt of %s.' % (self.name,
-                                                         self.period,
-                                                         dt))
-            if (abs(int(period/dt)*dt - period) >
-                    period * np.finfo(dt.dtype).eps):
-                raise NotImplementedError('The period of %s is %s, which is '
-                                          'not an integer multiple of its dt '
-                                          'of %s.' % (self.name,
-                                                      self.period,
-                                                      dt))
+                                                         self.period[:],
+                                                         dt*second))
+            # We check that the period is a multiple of dt. Because we cannot do
+            # any exact test, we use the following heuristic: we calculate all
+            # the multiples of the period that fit into 1000s and then check
+            # whether all these times correspond to the same time step as we
+            # obtain by first converting the period to the equivalent number of
+            # timesteps
+            if period < 1000:
+                n_periods = int(1000. / period)
+                periods = np.arange(n_periods) * period
+                if not np.all(timestep(periods*second, dt*second) % timestep(period*second, dt*second) == 0):
+                    raise NotImplementedError('The period of %s is %s, which is '
+                                              'not an integer multiple of its dt '
+                                              'of %s.' % (self.name,
+                                                          self.period[:],
+                                                          dt*second))
```

Comment:
How expensive is this check? In principle we could check just that last 10 or 100 values of ``periods`` and that's probably sufficient?